tremor-pipeline/src/op/trickle/select.rs

```diff
@@ -552,79 +599,32 @@ impl Operator for TrickleSelect {
         }
 
         let group_values: Vec<Value> = group_values.into_iter().map(Value::Array).collect();
+        let mut scratch = aggregates.clone(); // TODO: move to statement level to avoid overhead per event
```

Comment:
cloning aggregates on every emit seems extremely expensive

Reply:
This is on every event only :P
But yeah, this caused me lots of pain and will be fixed.